presto-spi/src/main/java/com/facebook/presto/spi/block/BlockEncodingBuffers.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.block;
+
+import io.airlift.slice.SliceOutput;
+
+import static java.lang.Math.max;
+
+public abstract class BlockEncodingBuffers
+{
+    protected int[] nextBatchOfPositions;
+    protected int nextBatchSize;
```

Comment:
I'd rename these to positions and positionCount.

Reply:
I renamed the positions. But I've seen OrcRecordReader uses nextBatchSize. I think nextBatchSize is more explanatory, and it avoids the confusions with the bufferedPositionCount